src/scripts/main.js

```diff
@@ -1,3 +1,317 @@
 'use strict';
 
-// write code here
+// 1. Implement table sorting by clicking on the title (in two directions).
+const table = document.querySelector('table');
+
+function getTableRows() {
+  return [...table.querySelectorAll('tbody tr')];
+}
+
+function mapTableData(rows) {
+  return rows.map((row) => {
+    return [...row.children].map((cell) => cell.innerText);
+  });
+}
+
+function getTableData() {
+  const rows = getTableRows();
+
+  return mapTableData(rows);
+}
+
+let sortDirection = true;
+
+function sortTable(columnIndex) {
+  const sortedData = sortTableData(columnIndex, sortDirection);
+
+  renderSortedTable(sortedData);
+  toggleSortDirection();
+}
+
+function sortTableData(columnIndex, direction) {
+  const data = getTableData();
+
+  return data.sort((rowA, rowB) => {
+    let cellA = rowA[columnIndex];
+    let cellB = rowB[columnIndex];
+
+    cellA = parseCellData(cellA, columnIndex);
+    cellB = parseCellData(cellB, columnIndex);
+
+    if (direction) {
+      return cellA > cellB ? 1 : -1;
+    }
+
+    return cellA > cellB ? -1 : 1;
+  });
```

Comment:
Performance issue: You are retrieving table data every time you sort it. It would be more efficient to retrieve the data once and then sort it as needed.